go/tao/keys_test.go

```diff
@@ -427,3 +428,54 @@ func TestNewOnDiskTaoSealedKeys(t *testing.T) {
 		t.Fatal("Couldn't read back a sealed, hosted keyset:", err)
 	}
 }
+
+// saveKeyset serializes and saves a Keys object to disk in plaintext.
+func saveKeyset(k *Keys) error {
+	cks, err := MarshalKeyset(k)
+	if err != nil {
+		return err
+	}
+	cks.Delegation = k.Delegation
+
+	m, err := proto.Marshal(cks)
+	if err != nil {
+		return err
+	}
+
+	if err = util.WritePath(k.SealedKeysetPath(), m, 0700, 0600); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Test genrating a new set of keys and saving/loading them to/from the disk unsealed.
+func TestUnsealedDelegatedKeysSaveLoad(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "TestInitHosted")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	defer os.RemoveAll(tempDir)
+
+	ft, err := NewSoftTao("", nil)
+	if err != nil {
+		t.Error("Couldn't initialize a SoftTao:", err)
+		return
+	}
+
+	k, err := NewTemporaryTaoDelegatedKeys(Signing|Crypting|Deriving, ft)
```

Comment:
Please also add a test of saving and reading a keyset that doesn't have a delegation